### PR TITLE
Add Makefile deployment.yaml and ansible deploy github action

### PR DESCRIPTION
**Pull request by mika018**
@bogdanadrianmarc Please also try building a running the image locally as part of the review.
@andrew-pickin-epi Just have a look through to see if there is anything obvious missing.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          image: ${{ steps.mapper.outputs.image }}/${{ steps.mapper.outputs.publish }}
+
+      - name: "Publish image"
+        if: ( steps.mapper.outputs.publish != '' && steps.gettag.outputs.tag != '' )
+        env:
+          IMAGE: ${{ steps.mapper.outputs.image }}/${{ steps.mapper.outputs.publish }}
+          TAG: ${{ steps.gettag.outputs.tag }}
+        shell: bash
+        run: |
+          make publish
```
Comment (andrew-pickin-epi) [inline .github/workflows/deploy.yml:75]: "master" has PAT

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	@docker run -p 3000:3000 --rm --name ppd_explorer -e RAILS_RELATIVE_URL_ROOT='' ${REPO}:${TAG}
+
+tag:
+	@echo ${TAG}
+
+test: assets
+	@./bin/rake test
+
+clean:
+	@./bin/rails assets:clobber
```
Comment (andrew-pickin-epi) [inline Makefile:41]: Just found elsewhere this fails if asset don't exist. I needed:
`[ -d public/assets ] && ./bin/rails assets:clobber || :`
confirm that `public/assets` is correct in your case.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+image: assets
+	@echo Building ${REPO}:${TAG} ...
+	@docker build --tag ${REPO}:${TAG} .
+	@echo Done.
+
+publish: image
+	@echo Publishing image: ${REPO}:${TAG} ...
+	@docker push ${REPO}:${TAG} 2>&1
+	@echo Done.
```
Comment (andrew-pickin-epi) [inline Makefile:22]: I think you need a `.dockerfile`. Probably a minimum of:
```
.dockerignore
.git/
.github/
.gitignore
CHANGELOG.md
Dockerfile
Makefile
README.md
deployment.yaml
```

**Review by andrew-pickin-epi — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (mika018) [reply .github/workflows/deploy.yml]: Added.

File: Makefile (hunk 2 above)
Comment (mika018) [reply Makefile]: Fixed.

File: Makefile (hunk 3 above)
Comment (mika018) [reply Makefile:22]: Added.

**Review by andrew-pickin-epi — APPROVED** [review]

**Review by bogdanadrianmarc — APPROVED** [review]
I went over the changes I made on the [original PR](https://github.com/epimorphics/ppd-explorer/pull/134) and this looks good in relation to that 👍

**Pull request merged**